splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java

```diff
@@ -129,9 +131,20 @@ private void configureHandler() throws Exception {
             // for the http producer
             .marshal().json(JsonLibrary.Jackson)
 
+            .setProperty("events_count", jsonpath("$.events.length()"))
+
+            // loops over events in the original message
+            .loop(exchangeProperty("events_count")).copy()
+            // picks one Event from the original message
+            .process(eventPicker)
+
             // Transform message to add splunk wrapper to the json
             .transform().simple("{\"source\": \"eventing\", \"sourcetype\": \"Insights event\", \"event\": ${body}}")
 
+            // aggregate transformed messages and append them together
+            .aggregate(constant(true), AggregationStrategies.bean(EventAppender.class))
```

Comment:
Note:
The first arg is a `correlationExpression` if I'm not mistaken. So any constant would do there.
https://camel.apache.org/components/3.16.x/eips/aggregate-eip.html

Reply:
:-)